File: backend/src/main/java/com/ngemail/domain/Email.java
package com.ngemail.domain;

import java.util.List;

public class Email {
    private String sender;
    private List<String> recipients;
    private List<String> ccs;
    private List<String> bccs;
    private String subject;
    private String content;

    public String getSender() {
        return sender;
    }

    public void setSender(String sender) {
        this.sender = sender;
    }

    public List<String> getRecipients() {
        return recipients;
    }

    public void setRecipients(List<String> recipients) {
        this.recipients = recipients;
    }

    public List<String> getCcs() {
        return ccs;
    }

    public void setCcs(List<String> ccs) {
        this.ccs = ccs;
    }

    public List<String> getBccs() {
        return bccs;
    }

    public void setBccs(List<String> bccs) {
        this.bccs = bccs;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    @Override
    public String toString() {
        final StringBuilder builder = new StringBuilder();
        builder.append("sender: " + getSender() + ", ");
        builder.append("recipients: " + (getRecipients() == null ? "null" : getRecipients().toString()) + ", ");
        builder.append("ccs: " + (getCcs() == null ? "null" : getCcs().toString()) + ", ");
        builder.append("bcc: " + (getBccs() == null ? "null" : getBccs().toString()) + ", ");
        builder.append("subject: " + getSubject() + ", ");
        builder.append("content: " + getContent() + ", ");
        return builder.toString();
    }
}

File: frontend/src/main/ng-email/src/app/model/email.ts
import { Mapper } from './mapper';
import { Person } from './person';

export class Email implements Mapper<Email>{
  constructor(
    private sender: Person = null,
    private recipients: Person[] = null,
    private content: string = null,
    private ccs?: Person[],
    private bccs?: Person[],
    private subject?: string) {}

  map(input: any): Email {
    let sender: Person;
    let recipients: Person[];
    let ccs: Person[];
    let bccs: Person[];
    let subject: string;
    let content: string;

    if (input.sender && input.sender[0]) {
      sender = input.sender[0];
    } else {
      throw new Error('Sender field is required');
    }

    if (input.recipients && input.recipients[0]) {
      recipients = input.recipients;
    } else {
      throw new Error('Recipients are required');
    }

    if (input.ccs) {
      ccs = input.ccs;
    }

    if (input.bccs) {
      bccs = input.bccs;
    }

    if (input.subject) {
      subject = input.subject.trim();
    }

    if (input.content) {
      content = input.content.trim();
    } else {
      throw new Error('Email content is required');
    }

    return new Email(sender, recipients, content, ccs, bccs, subject);
  }
}
